rust/src/lib/route.rs

```diff
@@ -77,6 +78,33 @@ impl Routes {
                     error!("{}", e);
                     return Err(e);
                 }
+                if let Some(dst) = route.destination.as_ref() {
+                    let dst_str = dst.as_str();
+                    if !is_ipv6_addr(dst_str) {
+                        let mut is_invalid_dst = false;
+                        if dst_str == "0.0.0.0" {
+                            is_invalid_dst = true;
+                        } else if dst_str.contains('/') {
+                            let net: Ipv4Net = dst.parse()?;
+                            if Ok(net.network()) == "0.0.0.0".parse()
+                                && net.prefix_len() >= 8
+                            {
+                                is_invalid_dst = true;
+                            }
+                        }
+                        if is_invalid_dst {
+                            let e = NmstateError::new(
+                                ErrorKind::InvalidArgument,
+                                "0.0.0.0/8 and its subnet cannot be used as \
+                            the route destination, please use the default \
+                            gateway 0.0.0.0/0 instead"
+                                    .to_string(),
+                            );
+                            error!("{}", e);
+                            return Err(e);
+                        }
```

Comment:
It seems a little bit complex IMO. I think you can skip the validation if the address is not in CIDR notation. In order to do that you can combine it with the `is_ipv6_addr` if statement. That means it will look something like that `if !is_ipv6_addr() && dst_str.contains('/')`. I think that is the right approach instead of having an extra variable to handle if the dst is invalid or not.